### PR TITLE
Run a docker container to flake8 the lambdas

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Some CI checking of our Python lambdas for consistency and to prevent simple errors reaching prod.

### Who is this change for?

xoi6zg

### Have the following been considered/are they needed?

- [ ] Deployed new versions

### CODE REVIEW

**Comment by alexwlchan** [comment]
👍 on the idea, but I think we can just install flake8 directly in Travis (as an install step):

```console
$ pip3 install flake8
```

I wonder whether the Docker container adds anything?

Separately, don’t forget that the pre-flight checks in `.travis.yml` will skip the build if there are changes in the Lambda directories but not anywhere else. You’ll need to change that.

**Comment by kenoir** [comment]
> I wonder whether the Docker container adds anything?

I think we should be moving in this direction as this makes our tests more consistent and portable.

**Comment by alexwlchan** [comment]
Also, can you push a flake8 violation to prove it breaks the build please? 😄

**Comment by kenoir** [comment]
@alexwlchan re re review pls!

File: docker/python3.6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.6
+
+VOLUME ["/data"]
+
+RUN pip install flake8
+
+COPY run.sh /app/run.sh
+
+ENTRYPOINT /app/run.sh
```
Comment (alexwlchan) [inline docker/python3.6/Dockerfile]: nit: I’d call this `flake8` instead of `python3.6`.

File: scripts/should_rebuild_master.py [hunk 2]
```diff
@@ -33,6 +33,10 @@ def should_rebuild_project(changed_files, project):
         print("*** Changes to the scripts directory mean we should rebuild")
         return True
 
+    elif any(f.startswith('.travis/') for f in changed_files):
```
Comment (alexwlchan) [inline scripts/should_rebuild_master.py]: I don't think you need the trailing slash – what about `.travis.yml`?

File: docker/python3.6/Dockerfile (hunk 1 above)
Comment (kenoir) [inline docker/python3.6/Dockerfile]: 👍 makes sense

File: scripts/should_rebuild_master.py (hunk 2 above)
Comment (kenoir) [inline scripts/should_rebuild_master.py]: Yup. I agree, as this doesn't regress functionality I would merge and then or another change.

**Pull request merged**